axelrod/strategies/titfortat.py

```diff
@@ -665,3 +665,42 @@ def strategy(self, opponent: Player) -> Action:
         if opponent.history[-1] == D:
             return D
         return C
+
+@FinalTransformer((D,), name_prefix=None)
+class EugineNier(Player):
+    """
+    Plays similar to Tit-for-Tat, but with two conditions:
+    1) Always Defect on Last Move
+    2) If other player defects five times, switch to all defects.
+
+    Names:
+
+    - EugineNier Strategy: [LessWrong2011]_
+    """
+
+    name = 'EugineNier'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': {'length'},
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.is_defector = False
+
+    def strategy(self, opponent: Player) -> Action:
+        if not self.history:
+            return C
+        if self.is_defector or opponent.history[-5:] == [D] * 5:
```

Comment:
I don't think it is actually, here's the source from http://lesswrong.com/lw/7f2/prisoners_dilemma_tournament_results/:

> Once the other player defects 5 times, switch to all defect.

So you can change this to be:

```python
699         if (not self.is_defector) and opponent.history.count(D) >= 5:
```

(Checking `not self.is_defector and` ensures we only count if it's not a Defector.)

Could you adjust the tests so that we check five defections (not necessarily consecutively):

```python
  643         # Becomes defector after 5 defections                                                                                                   
  644         opponent = axelrod.MockPlayer(actions=[D, C, D, D, D, D, C, C])                                                                         
  645         actions = [(C, D), (D, C), (C, D), (D, D),                                                                                              
  646                    (D, D), (D, D), (D, C), (D, C)]                                                                                              
  647         self.versus_test(opponent, expected_actions=actions)  
```

If you can please run those tests locally to make sure they work (I have just checked on my machine).